tests/HappyEyeBallsConnectionBuilderTest.php

```diff
@@ -302,8 +303,8 @@ public function testConnectWillStartConnectingWithAlternatingIPv6AndIPv4WhenReso
     {
         $timer = $this->getMockBuilder('React\EventLoop\TimerInterface')->getMock();
         $loop = $this->getMockBuilder('React\EventLoop\LoopInterface')->getMock();
-        $loop->expects($this->once())->method('addTimer')->with(0.1, $this->anything())->willReturn($timer);
-        $loop->expects($this->once())->method('cancelTimer')->with($timer);
+        $loop->expects($this->any())->method('addTimer')->with(0.1, $this->anything())->willReturn($timer);
+        $loop->expects($this->any())->method('cancelTimer')->with($timer);
```

Comment:
@clue changes like this one and similar changes in a few other files bug me as this indicates significant behavior changes in `react/promise` v3 that I wasn't expecting. What are your thoughts on it?

Reply:
Resolved as per https://github.com/reactphp/socket/pull/214#issuecomment-1227012613 via https://github.com/reactphp/promise/pull/229 :shipit: